lib/Document.php

```diff
@@ -221,14 +211,14 @@ public function createProperty($name, $value = null, array $parameters = null, $
         if (is_null($class)) {
             // If a VALUE parameter is supplied, we should use that.
             if (isset($parameters['VALUE'])) {
-                $class = $this->getClassNameForPropertyValue($parameters['VALUE']);
-                if (is_null($class)) {
-                    if (in_array(strtoupper($parameters['VALUE']), $this->allowedIllegalValues, true)) {
-                        unset($parameters['VALUE']);
-                        $class = $this->getClassNameForPropertyName($name);
-                    } else {
-                        throw new InvalidDataException('Unsupported VALUE parameter for '.$name.' property. You supplied "'.$parameters['VALUE'].'"');
-                    }
+                try {
+                    $class = $this->getClassNameForPropertyValue($parameters['VALUE']);
+                } catch (\Throwable $t) {
+                    $class = null;
+                }
+                if (is_null($class)) { // VALUE is malformed or illegal, drop it
+                    unset($parameters['VALUE']);
+                    $class = $this->getClassNameForPropertyName($name);
                 }
```

Comment:
Why not do this in the catch instead?

Reply:
I suppose because $this->getClassNameForPropertyValue() may return null?